x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/utils.ts

```diff
@@ -216,6 +216,81 @@ export function insertAtIndex<T>(array: T[], item: T, index: number): T[] {
   return [...array.slice(0, index), item, ...array.slice(index)];
 }
 
+/**
+ * Moves a contiguous block of steps (step + all descendants) up or down in the steps array.
+ * This means reordering around other contiguous blocks of steps.
+ * Maintains a strict index-based ordering of steps that reflects the hierarchy.
+ * @param stepRefs The flat array of StepActorRef
+ * @param stepId The customIdentifier of the root step to move
+ * @param direction 'up' or 'down'
+ * @returns A new reordered array of StepActorRef
+ */
+export function reorderSteps(
```

Comment:
no need for action, but even with the comments it's not so easy to understand what's going on here... Mapping the steps into a tree structure, then flattening it again is probably easier to follow